src/psp/guest_request.rs

```diff
@@ -0,0 +1,531 @@
+            ctx_req: ptr::null_mut(),
+            ctx_resp: ptr::null_mut(),
+        }
+    }
+
+    fn init(&mut self) -> bool {
+        self.req = SnpGuestRequestMsg::alloc_raw();
+        if self.req.is_null() {
+            prints!("ERROR: alloc_raw failed for the request\n");
+            return false;
+        }
+        self.resp = SnpGuestRequestMsg::alloc_raw();
+        if self.resp.is_null() {
+            prints!("ERROR: alloc_raw failed for the response\n");
+            return false;
+        }
+        true
+    }
+
+    fn free(&self) {
+        if !self.req.is_null() {
+            SnpGuestRequestMsg::free_raw(self.req);
+        }
+        if !self.resp.is_null() {
+            SnpGuestRequestMsg::free_raw(self.resp);
+        }
+        if !self.ctx_req.is_null() {
+            unsafe { EVP_CIPHER_CTX_free(self.ctx_req) };
+        }
+        if !self.ctx_resp.is_null() {
+            unsafe { EVP_CIPHER_CTX_free(self.ctx_resp) };
+        }
+    }
+
+    fn encrypt_request(&mut self, msg_type: u8, plaintext: &[u8]) -> bool {
+        let msg_seqno: u64 = match seqno_last_used().checked_add(1) {
+            Some(v) => v,
+            None => {
+                prints!("ERROR: Request sequence number overflow\n");
+                return false;
+            }
+        };
+        let aad_len: usize =
+            core::mem::size_of::<SnpGuestRequestMsgHdr>() - offset_of!(SnpGuestRequestMsgHdr, algo);
+        let mut iv = [0u8; DEFAULT_IV_SIZE];
+        iv[..U64_SIZE].copy_from_slice(&u64::to_ne_bytes(msg_seqno));
+        unsafe {
+            // Construct the request message header
+            (*self.req).hdr.hdr_sz = core::mem::size_of::<SnpGuestRequestMsgHdr>()
+                .try_into()
+                .unwrap();
+            (*self.req).hdr.algo = SNP_AEAD_AES_256_GCM;
+            (*self.req).hdr.hdr_version = HDR_VERSION;
+            (*self.req).hdr.msg_sz = core::mem::size_of_val(plaintext).try_into().unwrap();
+            (*self.req).hdr.msg_seqno = msg_seqno;
+            (*self.req).hdr.msg_type = msg_type;
+            (*self.req).hdr.msg_version = MSG_VERSION;
+            (*self.req).hdr.msg_vmpck = 0;
+            // Create new context
+            self.ctx_req = EVP_CIPHER_CTX_new();
+            if self.ctx_req.is_null() {
+                prints!("ERROR: Failed to get a new openssl CTX for encryption\n");
+                return false;
+            }
+            // Set encrypt operation
+            let mut ret: i32 = EVP_EncryptInit_ex(
+                self.ctx_req,
+                EVP_aes_256_gcm(),
+                ptr::null_mut(),
+                ptr::null(),
+                ptr::null(),
+            );
+            if ret != 1 {
+                prints!("ERROR: EVP_EncryptInit_ex failed ({})\n", ret);
+                return false;
+            }
+            // Provide key and iv lengths
+            ret = EVP_CIPHER_CTX_set_key_length(self.ctx_req, KEY_SIZE.try_into().unwrap());
```

Comment:
Is try_into() really needed here?  If it's because this should be a 32-bit value and KEY_SIZE is usize, I would say make KEY_SIZE a u32 and then do "as usize" where it is needed to be usize.